core/docker/build.sh

```diff
@@ -65,19 +65,23 @@ cp "$trino_client" "${WORK_DIR}/"
 tar -C "${WORK_DIR}" -xzf "${WORK_DIR}/trino-server-${TRINO_VERSION}.tar.gz"
 rm "${WORK_DIR}/trino-server-${TRINO_VERSION}.tar.gz"
 cp -R bin "${WORK_DIR}/trino-server-${TRINO_VERSION}"
-cp -R default "${WORK_DIR}/"
+mkdir -p "${WORK_DIR}/default"
+cp -R default/etc "${WORK_DIR}/default/"
 
 TAG_PREFIX="trino:${TRINO_VERSION}"
 
 for arch in "${ARCHITECTURES[@]}"; do
     echo "🫙  Building the image for $arch"
+    mkdir -p "${WORK_DIR}/default/apt/sources.list.d"
+    cp "default/apt/sources.list.d/mirrors-$arch.sources" "${WORK_DIR}/default/apt/sources.list.d/"
     docker build \
         "${WORK_DIR}" \
         --pull \
         --platform "linux/$arch" \
         -f Dockerfile \
         -t "${TAG_PREFIX}-$arch" \
         --build-arg "TRINO_VERSION=${TRINO_VERSION}"
+    rm -fr "${WORK_DIR}/default/apt/sources.list.d"
```

Comment:
why not keep not put those mirror files in `default/etc` and have all of those for build of images for each architecture. Would it break sth? I think the non-matching mirros woudl not be used. Then you do not need to modify this script at all.

Reply:
That was my initial approach, but if I remember correctly having all architectures defined at once made the `apt update` step longer since it tried to download package lists from all mirrors.

Reply:
One performance improvement that might be worth checking out is having all mirrors defined while also using `docker buildx` to build multiple architectures in parallel. That way we'd have a single shared build context, and the parallel build gains might offset some of the cost of fetching all mirror package lists.

Reply:
> One performance improvement that might be worth checking out is having all mirrors defined while also using `docker buildx` to build multiple architectures in parallel. That way we'd have a single shared build context, and the parallel build gains might offset some of the cost of fetching all mirror package lists.

I have a WIP branch with this approach in branch `michal/docker_ubuntu_mirrors_dockerx`